libs/mixer/melody.ts

```diff
@@ -250,7 +250,11 @@ namespace music {
                     if (!loop)
                         break
                 }
-                this.unregisterMelody();
+                // Unregister the melody if it was stopped and not restarted.
+                // (It was restarted if this._player is a new MelodyPlayer.)
+                if (!this._player) {
```

Comment:
```suggestion
                if (this._player == p) {
```
I think if it's null, it would already have been unregistered by stop(). Could you check that this works for you?

Reply:
That does seem to be the case, unregistering doesn't appear to be necessary for looping sounds

Instead, I moved the check to the (!loop) branch since non-looping melodies didn't get cleaned up when they ended.

Here's a test using a modified "music2.ts" including this change: https://makecode.com/_dLCTje3iXeMR